### PR TITLE
[MI-2360] Updated APIs to create/delete subscriptions and subscription card for "Build" and "Release" events

### PR DESCRIPTION
**Pull request by raghavaggarwal2308**
1. Updated the payload to create subscription with pipeline filters
2. Added filter chips on subscription card

Screenshot from 2022-11-18 16-15-34

3. Removed unused code

### CODE REVIEW

**Review by Kshitij-Katiyar — APPROVED** [review]

File: server/store/subscriptions.go [hunk 1]
```diff
@@ -58,7 +58,7 @@ func (subscriptionList *SubscriptionList) AddSubscription(userID string, subscri
 		subscriptionList.ByMattermostUserID[userID] = make(SubscriptionListMap)
 	}
 
-	subscriptionKey := GetSubscriptionKey(userID, subscription.ProjectName, subscription.ChannelID, subscription.EventType, subscription.Repository, subscription.TargetBranch, subscription.PullRequestCreatedBy, subscription.PullRequestReviewersContains, subscription.PushedBy, subscription.MergeResult, subscription.NotificationType, subscription.AreaPath)
+	subscriptionKey := GetSubscriptionKey(userID, subscription.ProjectName, subscription.ChannelID, subscription.EventType, subscription.Repository, subscription.TargetBranch, subscription.PullRequestCreatedBy, subscription.PullRequestReviewersContains, subscription.PushedBy, subscription.MergeResult, subscription.NotificationType, subscription.AreaPath, subscription.ReleasePipeline, subscription.BuildPipeline, subscription.BuildStatus, subscription.ApprovalType, subscription.ApprovalStatus, subscription.StageName, subscription.ReleaseStatus)
```
Comment (ayusht2810) [inline server/store/subscriptions.go:61]: do we really need all these in the subscription key?

File: server/plugin/api.go [hunk 2]
```diff
@@ -787,60 +833,6 @@ func (p *Plugin) handleGetUserAccountDetails(w http.ResponseWriter, r *http.Requ
...
-	organization := pathParams[constants.PathParamOrganization]
-	project := pathParams[constants.PathParamProject]
-
-	if len(strings.TrimSpace(organization)) == 0 || len(strings.TrimSpace(project)) == 0 {
-		p.API.LogError(constants.ErrorOrganizationOrProjectQueryParam)
-		p.handleError(w, r, &serializers.Error{Code: http.StatusBadRequest, Message: constants.ErrorOrganizationOrProjectQueryParam})
-		return
-	}
-
-	response, statusCode, err := p.Client.GetGitRepositories(organization, project, mattermostUserID)
```
Comment (ayusht2810) [inline server/plugin/api.go:803]: is this client function used anywhere else, if not we should remove it

File: server/plugin/api.go [hunk 3]
```diff
@@ -787,60 +833,6 @@ func (p *Plugin) handleGetUserAccountDetails(w http.ResponseWriter, r *http.Requ
...
-	project := strings.TrimSpace(pathParams[constants.PathParamProject])
-	repository := strings.TrimSpace(pathParams[constants.PathParamRepository])
-
-	if len(organization) == 0 || len(project) == 0 || len(repository) == 0 {
-		p.API.LogError(constants.ErrorRepositoryPathParam)
-		p.handleError(w, r, &serializers.Error{Code: http.StatusBadRequest, Message: constants.ErrorRepositoryPathParam})
-		return
-	}
-
-	response, statusCode, err := p.Client.GetGitRepositoryBranches(organization, project, repository, mattermostUserID)
```
Comment (ayusht2810) [inline server/plugin/api.go:827]: same (and for routes as well)

File: server/store/subscriptions.go (hunk 1 above)
Comment (raghavaggarwal2308) [reply server/store/subscriptions.go:61]: Yes, so that we can differentiate each subscription based on filters

**Review by ayusht2810 — APPROVED** [review]

File: server/store/utils.go [hunk 4]
```diff
@@ -136,8 +136,9 @@ func GetSubscriptionListMapKey() string {
 	return constants.SubscriptionPrefix
 }
 
-func GetSubscriptionKey(mattermostUserID, projectID, channelID, eventType, repository, targetBranch, pullRequestCreatedBy, pullRequestReviewersContains, pushedBy, mergeResult, notificationType, areaPath string) string {
-	return GetKeyMD5Hash(fmt.Sprintf("%s_%s_%s_%s_%s_%s_%s_%s_%s_%s_%s_%s", mattermostUserID, projectID, channelID, eventType, repository, targetBranch, pullRequestCreatedBy, pullRequestReviewersContains, pushedBy, mergeResult, notificationType, areaPath))
+func GetSubscriptionKey(mattermostUserID, projectID, channelID, eventType, repository, targetBranch, pullrequestCreatedBy, pullRequestReviewersContains, pushedBy, mergeResult, notificationType, areaPath, releasePipeline, buildPipeline, buildStatus, approvalType, approvalStatus, stageName, releaseStatus string) string {
+	// TODO: Optimize subscription key
```
Comment (avas27JTG) [inline server/store/utils.go:140]: I think this is not synced, pls sync when merging

**Review by avas27JTG — APPROVED** [review]

**Pull request merged**